applications/ULFapplication/custom_utilities/assign_point_neumann_conditions.h

```diff
@@ -86,45 +88,138 @@ namespace Kratos
 		{			
 			KRATOS_TRY;
-			KRATOS_WATCH("INSSSSSSSSSSSSSIIIIIIIIIIIIIIIiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiide of AssignPointNeumannConditions UTILITY")
+			KRATOS_WATCH("INSide of AssignPointNeumannConditions UTILITY")
+			const int TDim=ThisModelPart.ElementsBegin()->GetGeometry().WorkingSpaceDimension();
 
 			Properties::Pointer properties = ThisModelPart.GetMesh().pGetProperties(1);
 			int id = ThisModelPart.Conditions().size();
                         //const char* ConditionName = "NoSlipCondition2D";
-
+			KRATOS_WATCH(ThisModelPart.Conditions().size())
 			for(ModelPart::NodesContainerType::iterator it = ThisModelPart.NodesBegin(); 
 				it!=ThisModelPart.NodesEnd(); it++)
 			 {
 				if( it->FastGetSolutionStepValue(FLAG_VARIABLE) == 1.0)
 				   {	
-	      			//KRATOS_WATCH(">>>>>>>>>>>>>>>>>>>>> ASSIGNING THE NEUMANN POINT CONDITIONS -> EXTERNAL PRESSURE WILL BE APPLIED <<<<<<<<<<<<<<<<<<<<<");
+	      			       //KRATOS_WATCH(">>>>>>>>>>>>>>>>>>>>> ASSIGNING THE NEUMANN POINT CONDITIONS -> EXTERNAL PRESSURE WILL BE APPLIED <<<<<<<<<<<<<<<<<<<<<");
 				       Condition::NodesArrayType temp;	
 				       temp.reserve(1);
 				       temp.push_back(*(it.base()));	
 		
-									
-					Condition::Pointer p_cond = (KratosComponents<Condition>::Get("PointNeumann3D")).Create(id, temp, properties);	
-				
-					//Condition::Pointer p_cond = Condition::Pointer(new NoSlipCondition2D(id, temp, properties) );
-					(ThisModelPart.Conditions()).push_back(p_cond);
-
+					if (TDim==3)
+						{
+						Condition::Pointer p_cond = (KratosComponents<Condition>::Get("PointNeumann3D")).Create(id, temp, properties);	
+						(ThisModelPart.Conditions()).push_back(p_cond);
+						//KRATOS_WATCH(">>>>>>>>>>>>>>>>>>>>> ASSIGNING THE NEUMANN POINT CONDITIONS -> EXTERNAL PRESSURE WILL BE APPLIED <<<<<<<<<<<<<<<<<<<<<");
+						}
+					else if (TDim==2)
+						{
+						Condition::Pointer p_cond = (KratosComponents<Condition>::Get("PointNeumann2D")).Create(id, temp, properties);	
+						(ThisModelPart.Conditions()).push_back(p_cond);
+						
+						}
+					
 					id++;
+					
 				  }
 
 			}
+//KRATOS_WATCH(ThisModelPart.Conditions().size())
 			//KRATOS_WATCH("BEFORE SORTING")
 			ThisModelPart.Conditions().Sort();
+KRATOS_WATCH(ThisModelPart.Conditions().size())
+			
+
+			KRATOS_CATCH("")
+		}
+
+		void AssignPointNeumannConditionsDispAxisym(ModelPart& ThisModelPart)
+		{
+KRATOS_TRY;
+			KRATOS_WATCH("INSide of AssignPointNeumannConditions UTILITY")
+			const int TDim=ThisModelPart.ElementsBegin()->GetGeometry().WorkingSpaceDimension();
+
+			Properties::Pointer properties = ThisModelPart.GetMesh().pGetProperties(1);
+			int id = ThisModelPart.Conditions().size();
+                        //const char* ConditionName = "NoSlipCondition2D";
+			KRATOS_WATCH(ThisModelPart.Conditions().size())
+			for(ModelPart::NodesContainerType::iterator it = ThisModelPart.NodesBegin(); 
+				it!=ThisModelPart.NodesEnd(); it++)
+			 {
+			 	Condition::NodesArrayType temp;	
```

Comment:
it would be more efficient to create the array outside the loop and clear it in each iteration....

Reply:
corrected it as you requested